src/gc/gcinterface.h

```diff
@@ -444,6 +444,12 @@ class IGCHandleManager {
     virtual void DestroyHandleOfUnknownType(OBJECTHANDLE handle) = 0;
 
     virtual void* GetExtraInfoFromHandle(OBJECTHANDLE handle) = 0;
+
+    virtual void StoreObjectInHandle(OBJECTHANDLE handle, Object* object) = 0;
+
+    virtual bool StoreObjectInHandleIfNull(OBJECTHANDLE handle, Object* object) = 0;
+
+    virtual Object* CompareAndSwapObjectInHandle(OBJECTHANDLE handle, Object* object, Object* comparandObject) = 0;
```

Comment:
I felt that `InterlockedCompareExchangeObjectInHandle` was too Windows-specific (since that's the name of the actual Windows API that we use), while compare and swap is the generic term for the concept (https://en.wikipedia.org/wiki/Compare-and-swap).

If you feel strongly about this, I am happy to change it back.